src/map/script.c

```diff
@@ -7496,6 +7496,86 @@ BUILDIN(getelementofarray)
 	return true;
 }
 
+BUILDIN(array_find)
+{
+	struct script_data *data = script_getdata(st, 2);
+	struct reg_db *ref = NULL;
+	const char *name;
+	int i = 0, id, start_index, array_size;
+
+	if (!data_isreference(data)) {
+		ShowError("buildin_array_find: not a variable\n");
+		script->reportdata(data);
+		script_pushnil(st);
+		st->state = END;
+		return false;
+	}
+
+	start_index = reference_getindex(data);
+	if (start_index < 0 || start_index > SCRIPT_MAX_ARRAYSIZE) {
+		ShowError("buildin_array_find: starting index out of range (%d)\n", start_index);
```

Comment:
the script engine already ensures that `.@array[-1]` is illegal and that the index isn't above the max, so this check here is redundant and will never be triggered

Reply:
ok